bin/gftools-check-sandbox-family.py

```diff
@@ -0,0 +1,85 @@
+"""Make a diff/gif image of a site by swapping the GF production families
+with same families hosted on the sandbox server.
+
+
+Usage:
+gftools check-sandbox-family https://www.somesite.com
+"""
+import os
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options 
+from PIL import Image, ImageDraw, ImageFont
+import argparse
+from urllib.parse import urlsplit
+from io import BytesIO
+import time
+
+
+JS = """
+links = document.getElementsByTagName('link');
+for (i in links) {
+    if (typeof(links[i].href) != 'undefined') {
+         if (links[i].href.indexOf("fonts.googleapis.com") > -1)             
+             links[i].href = links[i].href.replace("fonts.googleapis", "fonts.sandbox.google");
+    }
+}
+"""
+WIDTH = 1024
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("url")
+    args = parser.parse_args()
+
+    chrome_options = Options()
+    chrome_options.add_argument("--headless")
+
+    with webdriver.Chrome(options=chrome_options) as driver:
+        driver.get(args.url)
+        required_height = driver.execute_script('return document.body.parentNode.scrollHeight')
+        driver.set_window_size(WIDTH, required_height)
+
+        try:
+            families_in_use = driver.find_elements_by_xpath(
+                    '//link[contains(@href, "fonts.googleapis.com/css")]'
+            )
+            for family in families_in_use:
+                print("Changing GF url %s to %s" % (
+                    family.get_attribute("href"), family.get_attribute("href").replace(
+                        "fonts.googleapis.com", "fonts.sandbox.google.com")
+                ))
+        except:
+            raise Exception("No hosted GF families found on %s" % args.url)
+
+        time.sleep(5)
+        before_img = driver.get_screenshot_as_png()
+        js = JS
+        driver.execute_script(js)
+        time.sleep(5)
+        after_img = driver.get_screenshot_as_png()
+
+        gif_path = urlsplit(args.url).netloc + ".gif"
+
+        with Image.open(BytesIO(before_img)) as before, Image.open(BytesIO(after_img)) as after:
+            font = ImageFont.truetype("/Library/Fonts/Arial.ttf", 32)
```

Comment:
This path is macOS specific, and AFAIK the rest of the script is not; it would be better to bundle a Roboto-Regular.ttf subset to ASCII and use that, I think.

Reply:
I'll use os.path so it's OS agnostic. Thanks.